content/recipes/necord.md

```diff
@@ -0,0 +1,365 @@
+import { Context, UserCommand, UserCommandContext, TargetUser } from 'necord';
+import { User } from 'discord.js';
+
+@Injectable()
+export class AppCommands {
+    @UserCommand({ name: 'Get avatar' })
+    public async getUserAvatar(
+        @Context() [interaction]: UserCommandContext,
+        @TargetUser() user: User
+    ) {
+        return interaction.reply({
+            embeds: [
+                new MessageEmbed()
+                    .setTitle(`Avatar ${user.username}`)
+                    .setImage(user.displayAvatarURL({ size: 4096, dynamic: true }))
+            ]
+        });
+    }
+}
+```
+
+#### Message Context Menu
+
+**Message commands** are application commands that appear on the context menu (right click or tap) of messages. They're a great way to surface quick actions for your app that target messages.
+
+```typescript
+@@filename(app.commands)
+import { Injectable } from '@nestjs/common';
+import { Context, MessageCommand, MessageCommandContext, TargetMessage } from 'necord';
+import { Message } from 'discord.js';
+
+@Injectable()
+export class AppCommands {
+    @MessageCommand({ name: 'Copy Message' })
+    public async copyMessage(
+        @Context() [interaction]: MessageCommandContext,
+        @TargetMessage() message: Message
+    ) {
+        return interaction.reply({ content: message.content });
+    }
+}
+```
+
+#### Buttons
+
+[Buttons](https://discord.com/developers/docs/interactions/message-components#buttons) are interactive components that render on messages. They can be clicked by users, and send an [interaction](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object) to your app when clicked.
+
+
+```typescript
+@@filename(app.components)
+import { Injectable } from '@nestjs/common';
+import { Context, Button, ButtonContext } from 'necord';
+
+@Injectable()
+export class AppComponents {
+    @Button('BUTTON')
+    public onButton(@Context() [interaction]: ButtonContext) {
+        return interaction.reply({ content: 'Button clicked!' });
+    }
+}
+```
+
+#### Select Menus
+
+[Select menus](https://discord.com/developers/docs/interactions/message-components#select-menus) are another interactive component that renders on messages. On desktop, clicking on a select menu opens a dropdown-style UI; on mobile, tapping a select menu opens up a half-sheet with the options.
+
+```typescript
+@@filename(app.components)
+import { Injectable } from '@nestjs/common';
+import { Context, StringSelect, StringSelectContext, Values } from 'necord';
+
+@Injectable()
+export class AppComponents {
+    @StringSelect('SELECT_MENU')
+    public onSelectMenu(@Context() [interaction]: StringSelectContext, @Values() values: string[]) {
+        return interaction.reply({ content: `Your selected color - ${values.join(' ')}` });
+    }
+}
+```
+
+All of built-in select menu components:
+- `@StringSelect`
+- `@UserSelect`
+- `@RoleSelect`
+- `@MentionableSelect`
+- `@ChannelSelect`
```

Comment:
I wonder if we being too descriptive here is needed. If the package change over time, we would have to update here as well... Instead, we could replace this with a link to the package's docs and only show general capabilities of the package